### ISSUE TITLE
BeanValidation for factories

### ISSUE THREAD

**Issue by andrus** [nhl/bootique#29]
Consider using JSR-349 (Bean Validation) for service factories loaded via ConfigurationFactory. Would be nice to declare property value expectations explicitly.

**Comment by fzakaria** [comment]
This is close to what dropwizard does.
Just wire it up in the configuration factory.

I prefer to use Typesafe, is my goto to simply provide my own ConfigurationFactory through an override module?

**Comment by andrus** [comment]
@fzakaria : yeah, Dropwizard was the inspiration here. Though we switched to other tasks and let this one linger. 

> I prefer to use Typesafe
Do you have links to this lib? The term is too general , so Google turns lots of unrelated stuff.

>  is my goto to simply provide my own ConfigurationFactory through an override module?
I'd say so. Or discuss an implementation and send us a PR? :)

**Comment by fzakaria** [comment]
Take a look at https://github.com/typesafehub/config

**Comment by andrus** [comment]
Thanks for the link. It looks like much more than just a property validator.
